dFusion/dFusionSpec.md

```diff
@@ -42,28 +45,45 @@ Accounts will pay "rent" to occupy an active account. The account index can be u
 Furthermore, we store a bi-map of token address to its index `0 <= t <= T`, for each token that can be traded on the exchange.
 When specifying tokens inside orders, deposits and withdrawel requests, we use the token's index `0 <= t <= T` instead of the full address.
 
-As orders, deposits and withdrawl requests are collected they are not directly stored in the smart contract.
-Doing so would require a `SSTORE` instruction which would be too gas-expensive.
+As limit orders, deposits and withdrawl requests are collected they are not directly stored in the smart contract.
+Doing so would require a `SSTORE` evm instruction for each item.
+This would be too gas-expensive:
 Assuming an order can be encoded in 256 bits, storing a batch of 10.000 on chain would cost ~5M gas (5.000 gas for SSTORE * 10.000 orders).
 Instead the smart contract emits a smart contract event containing the relevant order information (account, from_token, to_token, limit) and stores a rolling SHA hash.
 For a new order, the rolling hash is computed by hashing the previous rolling hash with the current order.
 
-// TODO: Pending withdrawels
+Any participants can apply pending deposit and withdrawl requests to the current account balance state.
+To do so, they provide a new state commitment that represents all account balances after the application of pending requests.
+Moreover, as the new state is stored on the smart contract, pending requests are reset.
+
+When a party applies withdrawl requests to the account balance state, they also provide the list of valid withdraws (in form of their merkle root) which we store in the smart contract inside a mapping (`transitionId` -> `valid withdraw merkle root`).
+Participants can later claim their withdraw by providing a merkle inclusion proof of their withdraw in any of the "valid withdraw merkle-roots".
+This will transfer their tokens from the smart contract's into their public address.
+In order to avoid double withdraws, we also store a bitmap for each "withdraw merkle-root".
+Each bit in that maps denotes if the withdraw has alreay been claimed.
+
+Participants can provide state transitions that apply pending deposits and withdrawls only while the order collection process is ongoing (the current batch is not yet frozen).
+Since price finding and order matching is a computationally expensive task, we don't want the account state to change while the optimization problem is ongoing, as this could potentially invalidate correct solutions (e.g. a withdraw could lead to insufficient balance for a matched trade).
```

Comment:
```suggestion
Since price finding and order matching is a computationally expensive task, the account state must not change while the optimization problem is ongoing, as this could potentially invalidate correct solutions (e.g. a withdraw could lead to insufficient balance for a matched trade).
```